packages/block-editor/src/hooks/gap.js

```diff
@@ -38,6 +39,45 @@ export function hasGapValue( props ) {
 	return props.attributes.style?.spacing?.blockGap !== undefined;
 }
 
+/**
+ * Returns a BoxControl object value from a given blockGap style.
+ * The string check is for backwards compatibility before Gutenberg supported
+ * split gap values (row and column) and the value was a string n + unit.
+ *
+ * @param {string? | Object?} rawBlockGapValue A style object.
+ * @return {Object?}                           A value to pass to the BoxControl component.
+ */
+export function getGapValueFromStyle( rawBlockGapValue ) {
+	if ( ! rawBlockGapValue ) {
+		return rawBlockGapValue;
+	}
+
+	const isValueString = typeof rawBlockGapValue === 'string';
+	return cleanEmptyObject( {
+		top: isValueString ? rawBlockGapValue : rawBlockGapValue?.top,
+		left: isValueString ? rawBlockGapValue : rawBlockGapValue?.left,
+	} );
```

Comment:
The style representation on the block attribute will be

```json
{"style":{"spacing":{"blockGap":{"top":"1.13em","left":"1.13em"}}}}
```

I'm wondering if `top` and `left` make sense long term, where `row` and `column` might be more understandable.

I know `blockGap` is a special case, but now might be the time to decide whether we want the properties to represent the CSS spec or whether we just roll with what we've got.

Reply:
Looking at it again, I see what you mean — `row` and `column` have a better semantic meaning, too, given that `top` and `left` don't make much sense in the context of the `gap` CSS property 🤔

One of the things that I like about `vertical` and `horizontal` is that it's the name used for the shorthand form of margin and padding when only two values are used, but as a label, it too is slightly removed from being aligned to the CSS spec. I think I might be leaning toward `row` and `column`?

But I suppose, either way, if we use `row` or `vertical`, when we generate the layout styles, we'll know which one to grab to apply it to `margin-top` in the default layout, and to `row-gap` (or the shorthand syntax) in the flex layout 😃

Reply:
Yes, naming is hard 😆 

I think `row` and `column` are more intuitive in this context, but it's hard to judge whether it'll remain so in the future and, as you point out, in conjunction with other layout usages.

I tend to view the layout abstraction as in flux and something perhaps we shouldn't prioritize when it comes to naming block gap properties. What I mean to say is that layout should be smart enough to know about its own implementation, and it might change anyway :)

I too am leaning towards `row` and `column` only because the rightly describe the properties, and will correspond (sorta) to how the rest of the internet describes gap.

We're using `vertical` and `horizontal` for the axial descriptors in block.json only because that what the BoxControl is looking for. 

We could extend BoxControl to accept other axial-like values such as `row` and `column` if folks think it'd be more consistent. 

I'd hate to set something in stone now and have to roll back later.